frontend/src/hooks/metrics.ts

```diff
@@ -0,0 +1,20 @@
+import { useIsLoggedIn } from "./session";
+import { useProfiles } from "./api/profile";
+
+export type MetricsState = "enabled" | "disabled" | "unknown";
+
+/**
+ * Should metrics be emitted?
+ * @returns MetricsState, 'unknown' if loading profile, 'enabled' or 'disabled' after loading
+ */
+export function useMetrics(): MetricsState {
+  const isLoggedIn = useIsLoggedIn();
+  const profileData = useProfiles();
+  if (isLoggedIn === "unknown") {
+    return "unknown";
+  }
+  const metricsEnabled =
+    isLoggedIn === "logged-out" ||
+    profileData?.data?.[0].metrics_enabled === true;
+  return metricsEnabled ? "enabled" : "disabled";
+}
```

Comment:
suggestion (blocking but I could be talked out of it): call `hasDoNotTrackEnabled` in here rather than in `_app.page.tsx`?